packages/uploadthing/test/adapters.test.ts

```diff
@@ -0,0 +1,301 @@
+          res: undefined;
+        }>(opts);
+        return {};
+      })
+      .onUploadComplete(uploadCompleteMock),
+  };
+
+  const eventHandler = createRouteHandler({
+    router,
+    config: {
+      uploadthingSecret: "sk_live_test",
+      fetch: mockExternalRequests,
+    },
+  });
+
+  it("gets H3Event in middleware args", async () => {
+    // FIXME: Didn't know how to declaratively create a H3Event to
+    // call the handler with directly, so I used the web-handler converter
+    // and sent in a Request and let H3 create one for me 🤷‍♂️
+    const res = await toWebHandler(createApp().use(eventHandler))(
+      new Request(createApiUrl("middleware", "upload"), {
+        method: "POST",
+        headers: baseHeaders,
+        body: JSON.stringify({
+          files: [{ name: "foo.txt", size: 48 }],
+        }),
+      }),
+    );
+    expect(res.status).toBe(200);
+
+    expect(middlewareMock.mock.calls[0][0].event).toBeInstanceOf(H3Event);
+    expect(middlewareMock.mock.calls[0][0].req).toBeUndefined();
+    expect(middlewareMock.mock.calls[0][0].res).toBeUndefined();
+
+    // Should proceed to have requested URLs
+    expect(fetchMock).toHaveBeenCalledTimes(1);
+    expect(fetchMock).toHaveBeenCalledWith(
+      "https://uploadthing.com/api/prepareUpload",
+      {
+        body: '{"files":[{"name":"foo.txt","size":48}],"routeConfig":{"blob":{"maxFileSize":"8MB","maxFileCount":1,"contentDisposition":"inline"}},"metadata":{},"callbackUrl":"https://not-used.com/","callbackSlug":"middleware"}',
+        headers: {
+          "Content-Type": "application/json",
+          "x-uploadthing-api-key": "sk_live_test",
+          "x-uploadthing-be-adapter": "h3",
+          "x-uploadthing-fe-package": "vitest",
+          "x-uploadthing-version": expect.stringMatching(/\d+\.\d+\.\d+/),
+        },
+        method: "POST",
+      },
+    );
+  });
+});
+
+describe("adapters:server", async () => {
+  const { createUploadthing, createRouteHandler } = await import(
+    "../src/server"
+  );
+  const f = createUploadthing();
+
+  const router = {
+    middleware: f({ blob: {} })
+      .middleware((opts) => {
+        middlewareMock(opts);
+        expectTypeOf<{
+          event: undefined;
+          req: Request;
+          res: undefined;
+        }>(opts);
+        return {};
+      })
+      .onUploadComplete(uploadCompleteMock),
+  };
+
+  const handlers = createRouteHandler({
+    router,
+    config: {
+      uploadthingSecret: "sk_live_test",
+      fetch: mockExternalRequests,
+    },
+  });
+
+  it("gets Request in middleware args", async () => {
+    const req = new Request(createApiUrl("middleware", "upload"), {
+      method: "POST",
+      headers: baseHeaders,
+      body: JSON.stringify({
+        files: [{ name: "foo.txt", size: 48 }],
+      }),
+    });
+    const res = await handlers.POST(req);
+    expect(res.status).toBe(200);
+
+    expect(middlewareMock.mock.calls[0][0].event).toBeUndefined();
```

Comment:
Is there a better way to get this than calls[0][0] eg. `toHaveBeenCalledWith()` or similar?

Reply:
not since i just wanna check undefined and instanceof :/

Reply:
i.e. i don't care to check the exact response / event as that'd get hella messy but i just wanna know if `opts.event` was an H3Event for example. unfortunately there isn't a

```ts
expect(middlewareMock).toHaveBeenCalledWith(
  expect.objectMatching({
    res: undefined,
    req: undefined,
    event: expect.instanceof(H3Event),
  })
);
```